src/components/Sep6/Sep6Withdraw.tsx

```diff
@@ -0,0 +1,287 @@
+  }
+
+  const formInitialState: FormData = {
+    withdrawType: {
+      type: "",
+    },
+    customerFields: {},
+    infoFields: {},
+  };
+
+  const [formData, setFormData] = useState<FormData>(formInitialState);
+  const dispatch = useDispatch();
+
+  const withdrawTypes = useMemo(
+    () => sep6WithdrawAsset.data.withdrawTypes?.types || { fields: {} },
+    [sep6WithdrawAsset],
+  );
+
+  const [activeWithdrawType, setActiveWithdrawType] = useState(
+    Object.keys(withdrawTypes)[0],
+  );
+
+  const withdrawTypesArr = useMemo(() => Object.entries(withdrawTypes), [
+    withdrawTypes,
+  ]);
+
+  useEffect(() => {
+    if (sep6WithdrawAsset.status === ActionStatus.NEEDS_INPUT) {
+      const initialWithdrawType = withdrawTypesArr[0][0];
+      setFormData({
+        withdrawType: {
+          type: initialWithdrawType,
+        },
+        customerFields: {},
+        infoFields: {},
+      });
+      setActiveWithdrawType(initialWithdrawType);
+    }
+  }, [sep6WithdrawAsset.status, withdrawTypesArr, dispatch]);
+
+  useEffect(() => {
+    if (sep6WithdrawAsset.status === ActionStatus.CAN_PROCEED) {
+      dispatch(sep6WithdrawAction());
+    }
+  }, [sep6WithdrawAsset.status, dispatch]);
+
+  const resetLocalState = () => {
+    setFormData(formInitialState);
+  };
+
+  const handleClose = () => {
+    dispatch(resetSep6WithdrawAction());
+    dispatch(resetActiveAssetAction());
+    resetLocalState();
+  };
+
+  const handleWithdrawTypeChange = (
+    event: React.ChangeEvent<HTMLSelectElement>,
+  ) => {
+    const { value } = event.target;
+    setActiveWithdrawType(value);
+    setFormData({
+      ...formData,
+      withdrawType: {
+        type: value,
+      },
+    });
+  };
+
+  const handleInfoFieldChange = (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    const { id, value } = event.target;
+
+    const updatedState = {
+      ...formData,
+      infoFields: {
+        ...formData.infoFields,
+        [id]: value,
+      },
+    };
+
+    setFormData(updatedState);
+  };
+
+  const handleCustomerFieldChange = (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    const { id, value } = event.target;
+
+    const updatedState = {
+      ...formData,
+      customerFields: {
+        ...formData.customerFields,
+        [id]: value,
+      },
+    };
+
+    setFormData(updatedState);
+  };
+
+  const handleSubmit = (
+    event: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+  ) => {
+    event.preventDefault();
+    dispatch(submitSep6DWithdrawFields({ ...formData }));
+  };
+
+  if (sep6WithdrawAsset.status === ActionStatus.NEEDS_INPUT) {
+    return (
+      <Modal visible={true} onClose={handleClose}>
+        <div className="ModalBody">
+          <Heading2
+            className="ModalHeading"
+            tooltipText={
+              <>
+                These are the fields the receiving anchor requires. The sending
+                client obtains them from the /info endpoint.{" "}
```

Comment:
I'm also differentiating between fields required by SEP-12 and fields required by SEP-6 because there are cases where SEP-12 and SEP-6 ask for the same fields and there doesn't appear to be an elegant way to de-dupe them